### PR TITLE
app: boards: add nucleo_h745zi_q support

### PR DESCRIPTION
**Pull request by KozhinovAlexander**
Add support nor NUCLEO-H745ZI_Q board.

### CODE REVIEW

File: app/boards/nucleo_h745zi_q.overlay
Comment (henrikbrixandersen) [inline app/boards/nucleo_h745zi_q.overlay:1]: Thanks! But please submit the basic CAN and USB enablement to the Zephyr repo instead, and leave only the CANnectivity-related configuration here.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: app/boards/nucleo_h745zi_q.overlay
Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:1]: Sure. I will do.

Comment (henrikbrixandersen) [reply app/boards/nucleo_h745zi_q.overlay:1]: Now that https://github.com/zephyrproject-rtos/zephyr/pull/80174 is merged, do you plan to pick up this PR again?

Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:1]: @henrikbrixandersen it is done - please take a look.

**Comment by KozhinovAlexander** [comment]
@henrikbrixandersen do you plan to take a look here?

File: app/boards/nucleo_h745zi_q.overlay [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+&zephyr_udc0 {
+	gs_usb0: gs_usb0 {
+		compatible = "gs_usb";
+	};
+};
+
+&fdcan1 {
+	/* Allocate half of the M_CAN message RAM to FDCAN1 */
+	reg = <0x4000a000 0x400>, <0x4000ac00 0x1400>;
+	reg-names = "m_can", "message_ram";
```
Comment (henrikbrixandersen) [inline app/boards/nucleo_h745zi_q.overlay:36]: Why is this needed? Why half?

File: app/boards/nucleo_h745zi_q.overlay [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	bosch,mram-cfg = <0x0 1 1 10 10 0 10 10>;
+	pinctrl-0 = <&fdcan1_rx_pb8 &fdcan1_tx_pb9>;
```
Comment (henrikbrixandersen) [inline app/boards/nucleo_h745zi_q.overlay:38]: How come this is changed from the board defaults of `pinctrl-0 = <&fdcan1_rx_pd0 &fdcan1_tx_pd1>;`?

File: app/boards/nucleo_h745zi_q.overlay [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	};
+};
+
+&arduino_i2c {
+	status = "disabled";
+};
+
+&arduino_serial {
+	status = "disabled";
+};
```
Comment (henrikbrixandersen) [inline app/boards/nucleo_h745zi_q.overlay:55]: This is not needed.

File: app/boards/nucleo_h745zi_q.overlay [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+	cannectivity: cannectivity {
+		compatible = "cannectivity";
+		timestamp-counter = <&counter2>;
+
+		channel0 {
+			compatible = "cannectivity-channel";
+			can-controller = <&fdcan1>;
+			state-gpios = <&gpiob 0 GPIO_ACTIVE_HIGH>;
+			activity-gpios = <&gpioe 1 GPIO_ACTIVE_HIGH>;
```
Comment (henrikbrixandersen) [inline app/boards/nucleo_h745zi_q.overlay:22]: So all this PR is really adding is support for a separate activity LED on this board (which does not have a designated CAN activity LED)?  If that is all, I see no reason to pull it in, sorry.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: app/boards/nucleo_h745zi_q.overlay (hunk 2 above)
Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:38]: There is only one shield, which patches perfectly the connector on nucleo_h745 board: https://www.waveshare.com/rs485-can-shield.htm and it needs this pinout.
The pd0 and pd1 pinse are default pins nucleo allocates for CAN accroding to their dicumentation. 
Therefore the change is needed.

File: app/boards/nucleo_h745zi_q.overlay (hunk 1 above)
Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:36]: Wil change to 10KiBi

File: app/boards/nucleo_h745zi_q.overlay (hunk 4 above)
Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:22]: True - the changes are very small. But they are needed to make CANnectivity running.
On one hand without this changes one needs to do them each time the board shall run CANNectivity.
On the other hand the introduced board is still a development board without specific CANnectivity functionality, therefore supporting CANnectivity needs this overlay.

File: app/boards/nucleo_h745zi_q.overlay (hunk 2 above)
Comment (henrikbrixandersen) [reply app/boards/nucleo_h745zi_q.overlay:38]: But just because you need to use this particular board with that particular shield does not make it a generic configuration suitable for other users of CANnectivity. Your change makes the configuration _incompatible_ with what others expect from the board schematic.

I suggest keeping this board + shield configuration in your own repository.

File: app/boards/nucleo_h745zi_q.overlay (hunk 4 above)
Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:22]: If I understand you right the lines 14 to 22 are not needed and CANnectivity will work with this board just right from Zephyr main repository?

File: app/boards/nucleo_h745zi_q.overlay (hunk 3 above)
Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:55]: I will drop it. Need to test first.

File: app/boards/nucleo_h745zi_q.overlay (hunk 2 above)
Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:38]: > But just because you need to use this particular board with that particular shield does not make it a generic configuration suitable for other users of CANnectivity. Your change makes the configuration _incompatible_ with what others expect from the board schematic.
> 
> I suggest keeping this board + shield configuration in your own repository.

It terms of CAN-L and CAN-H pins I may agree, but what about LEDs? see conversation: https://github.com/CANnectivity/cannectivity/pull/31#discussion_r1873082161

File: app/boards/nucleo_h745zi_q.overlay (hunk 4 above)
Comment (henrikbrixandersen) [reply app/boards/nucleo_h745zi_q.overlay:22]: > If I understand you right the lines 14 to 22 are not needed and CANnectivity will work with this board just right from Zephyr main repository?

None of the lines in this PR (except the bugfix for the missing `zephyr,canbus` chosen, which needs to go into Zephyr) are _needed_ for running CANnectivity on this board (have you tried)?

The addition of a separate activity LED is not really needed, as activity will be indicated on the state LED in it's absence. Separate activity LEDs are great for boards with dedicated CAN activity LEDs (e.g. boards intended for use as USB to CAN adapters). 

The only addition that _could_ warrant an overlay is really the added hardware timestamp counter, but I'm reluctant to start adding overlays for generic development boards only to add hardware timestamp support, a feature likely not many will require when using CANnectivity on a generic development board (or even know how to activate). These overlays will need to be maintained, tested - otherwise they will bitrot.

Comment (KozhinovAlexander) [reply app/boards/nucleo_h745zi_q.overlay:22]: Thank you.
On the other side I see problems with development boards in general - they are designed to cover as much applications as possible, therefore hardware conflicts are unavoidable. 
Sure, such conflicts may be solved by overlays, but the board is will still stay a development board and not a productive one.
I would suggest rejecting development board in general within CANnectivity.

**Comment by KozhinovAlexander** [comment]
Closed. Rejecting development boards approach in general shall be followed.
